src-input/duk_api_stack.c

```diff
@@ -3099,16 +3094,19 @@ DUK_LOCAL duk_ret_t duk__safe_to_stacktrace_raw(duk_hthread *thr, void *udata) {
 }
 
 DUK_EXTERNAL const char *duk_safe_to_stacktrace(duk_hthread *thr, duk_idx_t idx) {
-	const char *retval;
+	int rc;
 
 	DUK_ASSERT_API_ENTRY(thr);
 	idx = duk_require_normalize_index(thr, idx);
 
 	duk_dup(thr, idx);
-	(void) duk_safe_call(thr, duk__safe_to_stacktrace_raw, NULL /*udata*/, 1 /*nargs*/, 1 /*nrets*/);
-	retval = duk_safe_to_string(thr, -1);
+	rc = duk_safe_call(thr, duk__safe_to_stacktrace_raw, NULL /*udata*/, 1 /*nargs*/, 1 /*nrets*/);
+	if (rc != 0) {
+		duk_safe_to_stacktrace(thr, -1);
+	}
+	duk_replace(thr, -2);
```

Comment:
The replace here should be `duk_replace(thr, idx)`.

Reply:
Fixed!